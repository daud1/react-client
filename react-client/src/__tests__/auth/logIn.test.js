import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LogIn from '../../components/auth/logIn.component';
import moxios from 'moxios';


describe ('LogIn Component', () => {
    it('should render without crashing', () => {
    const wrapper = shallow(<LogIn />);
        expect(wrapper.exists(<div></div>)).toBe(true)
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<LogIn />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<LogIn />);
        
        wrapper.find('input[name="email"]')
            .simulate("change", { target: { value: 'user@mail.com' }  });
        wrapper.find('input[name="password"]')
            .simulate("change", { target: { value: 'password' }  });

        wrapper.find('form')
            .simulate('submit', { preventDefault() {} });

        moxios.stubRequest('http://localhost:5000/auth/login', {
            status: 200,
            response: {
                'MSG': 'Successfully Logged In!',
                'token': 'random_hexalpha_string',
            }
        });
        moxios.wait(() => {
            expect(wrapper.html()).toContain('Successfully Logged In');
            done();
        });

    });
})