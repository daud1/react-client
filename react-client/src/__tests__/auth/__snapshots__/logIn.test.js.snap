// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn Component matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      placeholder="Enter your email address"
      type="email"
    />
    <input
      name="password"
      placeholder="Enter your password"
      type="password"
    />
    <button
      type="submit"
    >
      Log In
    </button>
  </form>
</div>
`;
